xarray/tests/test_accessor_str.py

```diff
@@ -104,7 +104,7 @@ def test_contains(dtype) -> None:
     values = xr.DataArray(["Foo", "xYz", "fOOomMm__fOo", "MMM_"]).astype(dtype)
 
     # case insensitive using regex
-    pat = values.dtype.type("FOO|mmm")
+    pat = str
```

Comment:
I'm not sure what this was doing before I'm fairly sure we don't want this change

Reply:
```python
values = xr.DataArray(["Foo", "xYz", "fOOomMm__fOo", "MMM_"]).astype(np.str_)
values.dtype.type("FOO|mmm")
Out[44]: 'FOO|mmm'

values = xr.DataArray(["Foo", "xYz", "fOOomMm__fOo", "MMM_"]).astype(np.bytes_)
values.dtype.type("FOO|mmm")
Out[46]: b'FOO|mmm'
```

Looks like a bug. Probably trying to convert `type("FOO|mmm")` to `str` but doesn't take into account that it's a method and not the python function.

Reply:
Hmm, I think this was fixed in https://github.com/charliermarsh/ruff/pull/1962 (which went out in `v0.0.226`).

Reply:
I think this ran with 0.0.226, right @jhamman ?

```
https://github.com/charliermarsh/ruff-pre-commit@v0.0.226 (54.6KiB) (cached)
  sha: 6aa52c1faae70c9c500f47c60b66e00214d48f52
  envs:
  - python@python3 (10.4MiB) (cached)
```

Reply:
Actually, I think I ran `ruff --fix` with 0.0.146. I'll revert, update and try again.